src/components/customer-resource-show.js

```diff
@@ -70,10 +72,25 @@ export default class CustomerResourceShow extends Component {
     });
   };
 
+  // when there are no fields being edited this is equal to 0; when
+  // any number of fields are being edited this is greater than 0;
+  // this number is stored on the instance so that the state will only
+  // update when this count is equal to or greater than 0
+  editCount = 0;
+
+  handleEditing = (edit) => {
+    this.editCount = this.editCount + (edit ? 1 : -1);
+    let isEditing = this.editCount > 0;
+
+    if (isEditing !== this.state.isEditing) {
```

Comment:
Why `this.editCount > 0` and not just `!!this.editCount`?.  Is there a case where `editCount` could be negative?  

This sent me into some research on javascript's lack of a logical `xor`, because i thought it might be able to make the logic here more concise.  Long story short, you can use bitwise `xor` to make this a one-liner, but what you have is way clearer.  For entertainment's sake, here it is:

```
this.setState({ isEditing: !!(editCount ^= this.state.isEditing) })
```

Reply:
I was just being concise. It was `!!this.editCount` but I updated it after writing the comment above.

Nice use of a bitwise operator! I'll have to add that to my toolpouch. Another good one is:

```
if (~array.indexOf(value)) { ... }
```

`~` is falsy for `-1` but truthy for all other values